### DISCUSSION TITLE
Wasmer Runners

### DISCUSSION THREAD

**Discussion by syrusakbary** [wasmerio/wasmer#3997] [Ideas]
Right now Wasmer supports the WASI/WASIX, Emscripten and WCGI runners.

Would you like to see support for your custom games or ABIs in Wasmer? Please post here your use case and we will help you with it!

**Comment by DrRuhe** [comment]
Hi,
I am personally a big fan of reversible debugging. Sadly it is not as widespread as I would hope, partly because existing tools come with a lot of caveats. I was thinking if it is possible to create a WASM Runtime to act as a gdb-server with support for reverse debugging, where all the existing gdb-clients can connect to and then directly debug webassembly.  

This would allow directly debugging WASM, instead of having the runtime "in the way" of the debugger, where you are debugging both the runtime and the executing program.  

I think this could be done by creating a record/replay system for WASM and using it in a WASM runner. However as I am not familiar enough with wasmer, I cannot make an accurate assessment. I would be interested in implementing this, however, due to the complexity of such a feature paired with the deep understanding required for successful implementation, I would need guidance/mentoring.

**Comment by erikh** [comment]
I'm not sure, but I'm trying to create a load balancer with WASM plugins, for doing stuff like sharding and maxmind selection.

Is this where I need to be barking up? I'm a bit confused after quite a few trials with this in the rust/wasm ecosystem.

**Comment by mirzawaseembaig** [comment]
https://wordpress-4jtpt.wasmer.app/ how can i register the package
